react-carousel/my-app/src/Carousel.tsx

```diff
@@ -0,0 +1,48 @@
+import { ImageCard } from './ImageCard.tsx';
+import { NextButton } from './NextButton.tsx';
+import { PrevButton } from './PrevButton.tsx';
+import { Dots } from './Dots.tsx';
+import { useState, useEffect } from 'react';
+
+export type Image = {
+  src: string;
+  alt: string;
+};
+
+type Props = {
+  images: Image[];
+};
+
+export function Carousel({ images }: Props) {
+  const [currentIndex, setCurrentIndex] = useState(0);
+
+  useEffect(() => {
+    const timerId = setTimeout(() => {
+      setCurrentIndex((currentIndex + 1) % images.length);
+    }, 3000);
+    return () => clearTimeout(timerId);
+  }, [currentIndex, images.length]);
```

Comment:
`currentIndex` isn't a dependency in this function. On line 21, in your `setCurrentIndex` call, you are passing it an arrow function. That arrow function has one parameter, which you've named `currentIndex`. So the `currentIndex` variable on line 17 is not actually being used inside the effect function.

This is one reason I like to use `prev` when I use that form of the setter. I think it's less likely to confuse me!:
`setCurrentIndex((prev) => (prev + 1) % images.length);`